cmd/pj-rehearse/main.go

```diff
@@ -183,39 +191,60 @@ func rehearseMain() int {
 		metrics.RecordChangedCiopConfigs(changedCiopConfigData)
 	}
 
-	refs, chains, workflows, _, err := load.Registry(filepath.Join(o.releaseRepoPath, config.RegistryPath), false)
-	if err != nil {
-		logger.WithError(err).Error("could not load step registry")
-		return gracefulExit(o.noFail, misconfigurationOutput)
-	}
-	graph, err := registry.NewGraph(refs, chains, workflows)
-	if err != nil {
-		logger.WithError(err).Error("could not create step registry graph")
-		return gracefulExit(o.noFail, misconfigurationOutput)
-	}
-	changedRegistrySteps, err := config.GetChangedRegistrySteps(o.releaseRepoPath, jobSpec.Refs.BaseSHA, graph)
-	if err != nil {
-		logger.WithError(err).Error("could not get step registry differences")
-		return gracefulExit(o.noFail, misconfigurationOutput)
+	var changedRegistrySteps registry.NodeByName
+	var refs registry.ReferenceByName
+	var chains registry.ChainByName
+	var workflows registry.WorkflowByName
+	var graph registry.NodeByName
+
+	if !o.noRegistry {
+		refs, chains, workflows, _, err = load.Registry(filepath.Join(o.releaseRepoPath, config.RegistryPath), false)
+		if err != nil {
+			logger.WithError(err).Error("could not load step registry")
+			return gracefulExit(o.noFail, misconfigurationOutput)
+		}
+		graph, err = registry.NewGraph(refs, chains, workflows)
+		if err != nil {
+			logger.WithError(err).Error("could not create step registry graph")
+			return gracefulExit(o.noFail, misconfigurationOutput)
+		}
+		changedRegistrySteps, err = config.GetChangedRegistrySteps(o.releaseRepoPath, jobSpec.Refs.BaseSHA, graph)
+		if err != nil {
+			logger.WithError(err).Error("could not get step registry differences")
+			return gracefulExit(o.noFail, misconfigurationOutput)
+		}
+	} else {
+		graph, err = registry.NewGraph(refs, chains, workflows)
```

Comment:
This means that we create a graph out of nothing, as all of `refs`, `chains` and `workflows` are nil, are you sure that is what you want? If yes, please add a comment.

Reply:
And nit: The err check could reside outside of the branches

Reply:
Yeah, I basically need an empty graph, analogic to what I would get if I had an empty step-registry directory. The option should make rehearsal assume there were no changes in the registry (in that case, rehearsal tool does not need the registry at all). It's straightforward for templates and cluster profiles, not so much for registry unfortunately. 

For the error check, I think I can't extract it because in the first branch it's unfortunately in the middle and guards the `GetChangedRegistryStep()` from being called with possibly invalid graph.